test/Riok.Mapperly.Tests/Mapping/EnumerableTest.cs

```diff
@@ -848,102 +848,6 @@ public Task MapToReadOnlyCollectionProperty()
         return TestHelper.VerifyGenerator(source);
     }
 
-    [Fact]
-    public void EnumerableToReadonlyImmutableListShouldDiagnostic()
-    {
-        var source = TestSourceBuilder.Mapping(
-            "A",
-            "B",
-            "class A { public IEnumerable<int> Value { get; } }",
-            "class B { public System.Collections.Immutable.ImmutableList<int> Value { get; } }");
```

Comment:
IMO these test cases should still emit a warning. Having a matching source and target property, but the target property is read-only/immutable, is probably a bug in most cases.

Would still emit the warning by default. Users can ignore it by ignoring the target property via configuration

Reply:
Added a warning when trying to map to a readOnly immutable collection type.